### PR TITLE
Issue 21249 febeaccess

### LINKED ISSUE #21249: [Content] : Problems editing pages for Users with both FE and BE access
Reported via Support ticket: [105585](https://dotcms.zendesk.com/agent/tickets/105585)
Reproducible as of dotCMS 5.3.8.6, up to latest master (21.10)

When you edit a page in the dotCMS back-end, you can navigate to other pages in your Site -- actual HTML Pages or URL Mapped content -- and edit them right away. However, when your user has both `Back-End User` and `Front-End User` access, if you go to another HTML Page or URL Mapped content and try to edit it, you're editing the HTML Page you opened up from the **Site Browser** instead of the HTML Page or URL Mapped content you navigated to.

**To Reproduce**
1. Log into https://demo.dotcms.com/dotAdmin , go to the **Settings > Users** portlet, click the Admin User, un-check the `Front-End User` access box, and click **SAVE**.
2. Then, go to the **Site Browser**, and edit the `/index` page. The URL in the browser looks like this: https://demo.dotcms.com/dotAdmin/#/edit-page/content?url=%2Findex&language_id=1&host_id=48190c8c-42c4-46af-8d1a-0cd5db894797
3. On the page's main nav, click **DESTINATIONS**. The URL in the browser now changes to this, which is correct: https://demo.dotcms.com/dotAdmin/#/edit-page/content?url=%2Fdestinations%2Findex
4. Click the **EDIT** button to edit the current page. What you're now editing are the contents of the `/destinations/index` page. This is the expected behavior.
5. Now, go back to the **Settings > Users** portlet, click the Admin User, check the `Front-End User` access box, and click **SAVE**.
6. Go back to the **Site Browser**, and edit the `/index` page again. This is the URL in the browser: https://demo.dotcms.com/dotAdmin/#/edit-page/content?url=%2Findex&language_id=1&host_id=48190c8c-42c4-46af-8d1a-0cd5db894797
7. On the page's main nav, click **DESTINATIONS**. Notice that the URL in the browser **did not change at all this time**.
8. Click the **EDIT** button to edit the current page.
9. You're editing the contents of the `/index` page instead of the `/destinations/index` page, which is the page you clicked last time. This is **NOT** the expected behavior.

**Expected behavior**
Back-end Users should be able to click any link and edit the contents they see without problems.

**Additional context**
This issue seems to be related to the code changes from #20452 . After debugging the code with @freddyucv , we believe that the fix is to replace this line:
https://github.com/dotCMS/core/blob/40217a9e09aecb2715c97cbb2ecc3cd1c133b366/dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java#L58
With this one:
```java
final PageMode mode = userApi.getLoggedInUser(request) != null ? PageMode.getWithNavigateMode(request) : PageMode.setPageMode(request, PageMode.LIVE);
```
The only workaround we have found so far is to uncheck the `Front-End User` access checkbox for the conflicting Users.

### PR DESCRIPTION
**Pull request by jdotcms**
Fixes for edit mode access when users are FE & BE roles

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 1]
```diff
@@ -55,8 +75,9 @@ protected final void service(HttpServletRequest req, HttpServletResponse respons
                            : userApi.getAnonymousUserNoThrow();
         
         request.setRequestUri(uri);
-        final PageMode mode = user.isFrontendUser()?  PageMode.setPageMode(request, PageMode.LIVE) : PageMode.getWithNavigateMode(request);
-        
+        //final PageMode mode = user.isFrontendUser()? PageMode.setPageMode(request, PageMode.LIVE) : PageMode.getWithNavigateMode(request);
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java:78]: remove this commented code

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/util/PageModeTest.java [hunk 2]
```diff
@@ -51,6 +60,50 @@ private HttpServletRequest backEndRequest() {
...
+        session.setAttribute(com.dotmarketing.util.WebKeys.PAGE_MODE_SESSION, PageMode.NAVIGATE_EDIT_MODE);
+        request.setAttribute(WebKeys.USER, frontEndUser);
+        requestSession.setSession(session);
+        HttpServletRequestThreadLocal.INSTANCE.setRequest(request);
+        LoginMode.set(request, LoginMode.FE);
+        final PageMode pageMode = VelocityServlet.processPageMode(frontEndUser, frontEndRequest());
+        Assert.assertEquals(LoginMode.FE, LoginMode.get(request));
+        Assert.assertEquals(PageMode.LIVE, pageMode);
+    }
+
```
Comment (freddyucv) [inline dotCMS/src/integration-test/java/com/dotmarketing/util/PageModeTest.java:106]: Maybe you are missing some test here:

- When a user with FE and BE roles is logging in FE, the PageMode should be equals to LIVE
- When a user with FE and BE roles is logging in BE, the PageMode should be equals to NAVIGATE_EDIT_MODE

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 3]
```diff
@@ -40,6 +40,26 @@
...
+            return user.isFrontendUser()
+                    ? PageMode.setPageMode(request, PageMode.LIVE)
+                    :  user.isBackendUser()
+                    ? PageMode.getWithNavigateMode(request)
+                    :  PageMode.setPageMode(request, PageMode.LIVE);
+        }
+
+        return  LoginMode.FE == loginMode?
+                PageMode.setPageMode(request, PageMode.LIVE): PageMode.getWithNavigateMode(request);
+    }
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java:62]: should this method be here?, maybe it should be move to PageMode

File: dotCMS/src/main/java/com/dotmarketing/util/LoginMode.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    UNKNOWN, BE, FE;
+
+    public static LoginMode get() {
+
+        final HttpServletRequest req = Try.of(()-> HttpServletRequestThreadLocal.INSTANCE.getRequest()).getOrNull();
+        return null == req? UNKNOWN: get(req);
+
+    }
+
+    public static LoginMode get(final HttpServletRequest request) {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/util/LoginMode.java:26]: doc

File: dotCMS/src/main/java/com/dotmarketing/util/LoginMode.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+            final LoginMode loginMode = (LoginMode) request.getSession(false)
+                    .getAttribute(WebKeys.LOGIN_MODE_PARAMETER);
+            return loginMode;
+        }
+
+        return UNKNOWN;
+    }
+
+    public static void set(final HttpServletRequest request, final LoginMode loginMode) {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/util/LoginMode.java:39]: doc?

**Review by freddyucv — COMMENTED** [review]
Maybe we should include some test here 

https://github.com/dotCMS/core/blob/7b5b318fa7b5c89804aead937bb1709766c24c91/dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java#L63

To test rendering a page with a user with both roles (FE and BE)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 3 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java:62]: I thought the same, but I am concern about breaking something

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 1 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java:78]: done

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**